src/Database/Type/EnumType.php

```diff
@@ -0,0 +1,185 @@
+     * @return string|int|null
+     */
+    public function toDatabase(mixed $value, Driver $driver): string|int|null
+    {
+        if ($value === null) {
+            return null;
+        }
+
+        if ($value instanceof BackedEnum) {
+            return $value->value;
+        }
+
+        if (is_int($value) || is_string($value)) {
+            return $value;
+        }
+
+        throw new InvalidArgumentException(sprintf(
+            'Cannot convert value of type `%s` to string or integer',
+            get_debug_type($value)
+        ));
+    }
+
+    /**
+     * Transform DB value to backed enum instance
+     *
+     * @param mixed $value The value to convert.
+     * @param \Cake\Database\Driver $driver The driver instance to convert with.
+     * @return \BackedEnum|null
+     */
+    public function toPHP(mixed $value, Driver $driver): ?BackedEnum
+    {
+        if ($value === null) {
+            return null;
+        }
+
+        if (get_debug_type($value) !== $this->backingType) {
+            throw new InvalidArgumentException(sprintf(
+                'Given value type `%s` does not match associated `%s` backed enum',
+                get_debug_type($value),
+                $this->backingType
+            ));
+        }
+
+        return $this->enumClassName::tryFrom($value);
+    }
+
+    /**
+     * Get the correct PDO binding type for string or integer data.
+     *
+     * @param mixed $value The value being bound.
+     * @param \Cake\Database\Driver $driver The driver.
+     * @return int
+     */
+    public function toStatement(mixed $value, Driver $driver): int
+    {
+        if (is_int($value)) {
+            return PDO::PARAM_INT;
+        }
+
+        return PDO::PARAM_STR;
+    }
+
+    /**
+     * Marshals request data
+     *
+     * @param mixed $value The value to convert.
+     * @return \BackedEnum|null Converted value.
+     */
+    public function marshal(mixed $value): ?BackedEnum
+    {
+        if ($value === null) {
+            return null;
+        }
+
+        if ($value instanceof $this->enumClassName) {
+            return $value;
+        }
+
+        if (get_debug_type($value) !== $this->backingType) {
+            throw new InvalidArgumentException(sprintf(
+                'Given value type `%s` does not match associated `%s` backed enum',
+                get_debug_type($value),
+                $this->backingType
+            ));
+        }
+
+        $enumInstance = $this->enumClassName::tryFrom($value);
+        if ($enumInstance === null) {
+            throw new InvalidArgumentException(sprintf(
+                'Given value `%s` is not present inside associated `%s` backed enum',
+                $value,
+                $this->enumClassName
+            ));
+        }
+
+        return $enumInstance;
+    }
+
+    /**
+     * @param string $enumClassName The enum class name
+     * @return string
+     */
```

Comment:
```suggestion
    /**
     * Create an `EnumType` that is paired with the provided `$enumClassName`.
     *
     * ### Usage
     *
     * ```
     * // In a table class
     * $this->getSchema()->setColumnType('status', EnumType::for(StatusEnum::class));
     * ```
     *   
     * @param string $enumClassName The enum class name
     * @return string
     */
```